app/gui/view/component-browser/searcher-list-panel/src/lib.rs

```diff
@@ -0,0 +1,633 @@
+        self.label.set_font(style.section.heading.font.clone());
+        self.label.set_position_y(-0.75 * style.section.heading.size.raw);
+        self.label.set_position_x(3.0 + style.content.padding);
+    }
+}
+
+impl<T: ObjectOps + CloneRef> LabeledSection<T> {
+    fn set_parent(&self, parent: impl ObjectOps) {
+        parent.add_child(&self.content);
+        parent.add_child(&self.label);
+        parent.add_child(&self.divider);
+    }
+}
+
+/// Trait that provides functionality for layouting and layer setting for structs used in the
+/// `LabeledSection`.
+trait SectionContent {
+    fn set_layers(&self, layers: &Layers, scroll_layer: &Layer);
+    fn height(&self) -> f32;
+    fn set_position_top_y(&self, position_y: f32);
+}
+
+impl SectionContent for component_group::wide::View {
+    fn set_layers(&self, layers: &Layers, _scroll_layer: &Layer) {
+        self.model().set_layers(layers);
+    }
+
+    fn height(&self) -> f32 {
+        self.size.value().y
+    }
+
+    fn set_position_top_y(&self, position_y: f32) {
+        self.set_position_y(position_y - self.height() / 2.0);
+    }
+}
+
+impl SectionContent for column_grid::ColumnGrid {
+    fn set_layers(&self, layers: &Layers, scroll_layer: &Layer) {
+        self.model().set_layers(layers, scroll_layer);
+    }
+
+    fn height(&self) -> f32 {
+        self.size.value().y
+    }
+
+    fn set_position_top_y(&self, position_y: f32) {
+        self.set_position_y(position_y);
+    }
+}
+
+impl<T: SectionContent + CloneRef> LabeledSection<T> {
+    fn set_layers(&self, layers: &Layers, scroll_layer: &Layer) {
+        self.content.set_layers(layers, scroll_layer);
+        scroll_layer.add_exclusive(&self.label);
+        self.label.add_to_scene_layer(scroll_layer);
+        scroll_layer.add_exclusive(&self.divider);
+    }
+
+    /// Full height of the section including header.
+    fn height(&self, style: &Style) -> f32 {
+        let label_height = style.section.heading.height();
+        let body_height = self.content.height();
+        let next_section_offset = 17.0; // TODO[MM] this should not be needed.
+        body_height + label_height + next_section_offset
+    }
+
+    /// Set the top y position of the section.
+    fn set_base_position_y(&self, position_y: f32, style: &Style) {
+        match SECTION_HEADER_PLACEMENT {
+            SectionHeaderPlacement::Top => {
+                self.label.set_position_y(position_y - self.label.height.value() / 1.5);
+                self.divider.set_position_y(position_y);
+                let content_position_y =
+                    position_y - style.section.heading.offset - style.section.heading.height();
+                self.content.set_position_top_y(content_position_y);
+            }
+            SectionHeaderPlacement::Bottom => {
+                self.label.set_position_y(
+                    position_y - self.content.height() - self.label.height.value() / 1.5,
+                );
```

Comment:
such 3-line expressions can be made 2-liners by extracting values to variables. Moreover, they will be easier to read this way.